standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java

```diff
@@ -461,6 +459,14 @@ public Thread newThread(@NotNull Runnable r) {
     // TODO: AcceptQueueSize needs to be higher for HMS
     connector.setAcceptQueueSize(maxWorkerThreads);
     // TODO: Connection keepalive configuration?
+    connector.addBean(new HttpChannel.Listener() {
+      @Override
+      public void onComplete(Request request) {
+        LOG.debug("Request " + request + " is completed. HttpChannel was: " + request.getHttpChannel()
+            + ", HttpTransport: " + request.getHttpChannel().getHttpTransport());
+        HMSHandler.cleanupHandlerContext();
```

Comment:
Is it possible to add a unit test to make sure we indeed clean up all of objects that need to be removed?

Reply:
`testJDOPersistanceManagerCleanup` in TestHiveMetaStore is testing that. The failure in that test led to the fixes proposed in this patch. Without this patch, testJDOPersistanceManagerCleanup() fails in Thrift over HTTP mode.

Reply:
should we track the open connection metric as well?

Reply:
@dengzhhu653 : Yes we should. Since that is unrelated to this patch, can we take it up in a separate task ?

Reply:
My idea is that this is a relative minor change, we can do it as a plus, igore this if you already have a plan for the metric.